### DISCUSSION TITLE
When a scenario is triggered, another related scenario can't be triggered

### DISCUSSION THREAD

**Discussion by marioekaputta** [bahasa-ai/otomatis-guidance#14] [Show and tell]
I created 2 scenarios for Zendesk & Tokopedia demo (When Chat/Discussions are received from the customer in Tokopedia, Send a reply from Zendesk).

-  When Incoming Message from Tokopedia, Show It on Zendesk

This is the first scenario outline:
![image](https://user-images.githubusercontent.com/71392755/129179823-e593ca52-c8fa-4f73-b16e-7101c01c3b26.png)

- When New Comment by Agent on Zendesk, Send Text Reply to Tokopedia

The outline: 
![image](https://user-images.githubusercontent.com/71392755/129180022-4baf5974-bac2-46a2-a425-4e2089b0a0fb.png)

Both of the scenarios work just fine during the development and when I run it alone, but when both of them runs together, there are a few problems like:

1. Duplicate message
![image](https://user-images.githubusercontent.com/71392755/129180762-a26731b2-a9dd-4d13-9f14-0702e10599c2.png)

2. The customer's message on Tokopedia was sent back to them
<img width="950" alt="Screen Shot 2021-08-12 at 15 35 53" src="https://user-images.githubusercontent.com/71392755/129180837-3db353c9-a4e2-452f-8967-16af51fc7e19.png">

3. After running the first scenario and the second, **sometimes** the first can't run again

**Comment by marioekaputta** [comment]
[RESOLVED]

I try to troubleshoot the problems, and I figure out what went wrong.

1. Duplicate message
Basically, this one is the simplest one. Two accounts turn on the scenario at the same time.

2. The customer's message on Tokopedia was sent back to them
This one is pretty tricky, but I figure out if scenario 2 (When New Comment by Agent on Zendesk, Send Text Reply to Tokopedia) is also triggered when I send messages from Tokopedia. It turns out the triggered condition is set if there are any new comments/updates on the ticket not only from the agent. That's why when scenario 1 done the job, it also triggered scenario 2. I haven't figure out a way to identify the comment's author, but I can identify the media to comment on the ticket so I add the condition below. **Note:** The problem about the condition below is if the end-user sends a comment to this ticket via other apps like WA/Twitter then this scenario will be triggered. Still working on this
![image](https://user-images.githubusercontent.com/71392755/129181239-131e0430-5862-443c-b3ef-8dfa228c71a6.png)

3.  After running the first scenario and the second, **sometimes** the first can't run again.
It turns out that the scenario always running but the path is not as we expected. When users do some test on this scenario, usually they will create a new user and new ticket (Path non-existing ticket). **If the user didn't delete the results of the tests they created, this will lead to confusion for action Find Ticket**. The action Find Ticket will return more than one ticket and I'm not sure but I think the order is not always the same so the message will sometimes go back to the ticket we expected or the other one we didn't know.

However, there is a case when the scenario wasn't triggered at all. I haven't figure out about this yet, but it's not happening again since I figure out the problem above.

**Reply by mgilangjanuar** [reply]
Cool! 🎊
